src/components/graph/network-modification-tree-model.ts

```diff
@@ -174,29 +179,31 @@ export default class NetworkModificationTreeModel {
         });
     }
 
-    updateNodes(updatedNodes: (NetworkModificationNode | RootNode)[]) {
+    updateNodes(updatedNodes: (NetworkModificationNodeData | RootNodeData)[]) {
         updatedNodes.forEach((node) => {
             const indexModifiedNode = this.treeNodes.findIndex((othernode) => othernode.id === node.id);
             if (indexModifiedNode !== -1) {
-                const globalBuildStatus = isNetworkModificationNode(node)
-                    ? node.nodeBuildStatus?.globalBuildStatus
-                    : undefined;
-                const localBuildStatus = isNetworkModificationNode(node)
-                    ? node.nodeBuildStatus?.localBuildStatus
-                    : undefined;
-                this.treeNodes[indexModifiedNode].data = {
-                    ...this.treeNodes[indexModifiedNode].data,
-                    label: node.name,
-                    globalBuildStatus: globalBuildStatus,
-                    localBuildStatus: localBuildStatus,
-                    readOnly: node.readOnly,
+                const nodeToUpdate = this.treeNodes[indexModifiedNode];
+                const modificationNodeData = getModificationNodeDataOrUndefined(node);
+                const globalBuildStatus = modificationNodeData?.nodeBuildStatus?.globalBuildStatus;
+                const localBuildStatus = modificationNodeData?.nodeBuildStatus?.localBuildStatus;
+                this.treeNodes[indexModifiedNode] = {
+                    ...nodeToUpdate,
+                    data: {
+                        ...this.treeNodes[indexModifiedNode].data,
+                        label: node.name,
+                        globalBuildStatus: globalBuildStatus,
+                        // @ts-ignore TODO separate update root and update modification ?
```

Comment:
separate root and modification `node`

Reply:
ts-ignore removed i separated the updates